rpctest/node.go

```diff
@@ -180,46 +211,115 @@ func newNode(config *nodeConfig, dataDir string) *node {
 // test case, or panic, it is important that the process be stopped via stop(),
 // otherwise, it will persist unless explicitly killed.
 func (n *node) start() error {
+	var err error
+
+	// Redirect stderr.
+	n.stderr, err = n.cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	n.wg.Add(1)
+	go func() {
+		defer n.wg.Done()
+		r := bufio.NewReader(n.stderr)
+		for {
+			line, err := r.ReadBytes('\n')
+			if err == io.EOF {
+				n.tracef("stderr: EOF")
+				return
+			}
+			// It's ok to race n.pid.
```

Comment:
Races are actually never ok because they can be used access arbitrary memory.

Reply:
It really is in this case but I'll perform some magic. This will require a really crappy fix.

We'll need to launch the process WITHOUT logging in the pipe readers.

Reply:
It doesn't directly apply to this case, but a good article that showcases one way that a race can be used to access arbitrary memory is https://research.swtch.com/gorace.

Also, you can check the official Go memory model documentation at https://golang.org/ref/mem (courtesy of @jrick).

Reply:
I added a waitgroup to handle this.

I am pretty familiar with both those links :-)